src/main/java/org/gridsuite/shortcircuit/server/utils/FeederResultSpecifications.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.shortcircuit.server.utils;
+
+import org.gridsuite.shortcircuit.server.dto.Filter;
+import org.gridsuite.shortcircuit.server.entities.FeederResultEntity;
+import org.gridsuite.shortcircuit.server.entities.ShortCircuitAnalysisResultEntity;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+import java.util.UUID;
+
+public final class FeederResultSpecifications {
+
+    // Utility class, so no constructor
+    private FeederResultSpecifications() {
+    }
+
+    public static Specification<FeederResultEntity> resultUuidEquals(UUID value) {
+        return (feederResult, cq, cb) -> cb.equal(feederResult.get("faultResult").get("result").get("resultUuid"), value);
+    }
+
+    public static Specification<FeederResultEntity> contains(String column, String value) {
+        return (feederResult, cq, cb) -> cb.like(feederResult.get(column), "%" + value + "%");
+    }
+
+    public static Specification<FeederResultEntity> startsWith(String column, String value) {
+        return (feederResult, cq, cb) -> cb.like(feederResult.get(column), value + "%");
+    }
+
+    public static Specification<FeederResultEntity> notEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.notEqual(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> lessThanOrEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.lessThanOrEqualTo(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> greaterThanOrEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.greaterThanOrEqualTo(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> buildSpecification(ShortCircuitAnalysisResultEntity resultEntity, List<Filter> filters) {
```

Comment:
pass the uuid directly instead of the entity ? this prevents mistakes like loading all the fault results or other child entities with the accesssors

Reply:
You mean writing 
` Specification<FeederResultEntity> specification = FeederResultSpecifications.buildSpecification(result.get().getResultUuid(), filters);
`
instead of calling the getter in buildSpecification ?

Reply:
yes

Reply:
Done